### ISSUE TITLE
argo terminate marks task as Succeeded instead of Failed

### ISSUE THREAD

**Issue by stephpalis** [argoproj/argo-workflows#9209]
## Checklist

 

* [ ] Double-checked my configuration.
* [ ] Tested using the latest version.
* [ ] Used the Emissary executor.

## Summary

What happened/what you expected to happen?
* When running `argo terminate` on a running workflow, it properly terminates the workflow, but marks the task as `Succeeded`. I would expect the workflow to be marked as `Failed`.

What version are you running?
* Operator: `v3.3.8` 
* ArgoCLI: `v3.1.0-rc13 `

## Diagnostics

* Launch a workflow
* While it is still running, call argo terminate on the workflow
* The workflow will be marked as succeeded

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
Could you check if your step is a retried step? Also what's the status of the previous step?

**Comment by stephpalis** [comment]
Hey @terrytangyuan - I'm not quite sure how to reproduce the issue. The pod was in a pending state when I saw the bug occur, but it seems to work as expected when the pods are in a running state.

**Comment by DiegoFlop** [comment]
I have the same problem, but I don't now how reproduce the issue.

My problem is, that when I Terminate a workflow sometimes the workflow pipeline is marked as success but see the pipeline in UI and I see that all next steps to pipeline since that I terminated workflow are omitted and marked as success.

![Captura de pantalla 2022-07-22 a las 22 24 37](https://user-images.githubusercontent.com/34077334/180542415-c252f593-b499-4ccb-9f44-88ca200708b3.png)

In this picture it can see that problem. I terminated workflow in install_dependencies, however all next steps are marked success immediately
